plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GestureState.kt

```diff
@@ -0,0 +1,56 @@
+package com.mapbox.maps.plugin.gestures
+
+import com.mapbox.android.gestures.AndroidGesturesManager
+
+internal class GestureState(private val gesturesManager: AndroidGesturesManager) {
+  enum class Type {
+    DoubleTap,
+    Scale,
+    ScaleQuickZoom,
+    Shove,
+  }
+
+  private val rememberedGestureSettings = mutableMapOf<Type, Boolean>()
+
+  fun saveAndDisable(gesture: Type): Boolean {
+    return when (gesture) {
+      Type.DoubleTap -> {
+        gesturesManager.moveGestureDetector.isEnabled.also {
+          gesturesManager.moveGestureDetector.isEnabled = false
+        }
+      }
+      Type.Scale -> {
+        gesturesManager.rotateGestureDetector.isEnabled.also {
+          gesturesManager.rotateGestureDetector.isEnabled = false
+        }
+      }
+      Type.ScaleQuickZoom -> {
+        gesturesManager.moveGestureDetector.isEnabled.also {
+          gesturesManager.moveGestureDetector.isEnabled = false
+        }
+      }
+      Type.Shove -> {
+        gesturesManager.moveGestureDetector.isEnabled.also {
+          gesturesManager.moveGestureDetector.isEnabled = false
+        }
+      }
```

Comment:
The function name is a bit confusing, the implementation for type `DoubleTap`, `ScaleQuickZoom`, `Shove` are exactly the same, shall we maybe only have two types, e.g.
```
  fun saveAndDisable(gesture: Type): Boolean {
    return when (gesture) {
      Type.Move -> {
        gesturesManager.moveGestureDetector.isEnabled.also {
          gesturesManager.moveGestureDetector.isEnabled = false
        }
      }
      Type.Rotate -> {
        gesturesManager.rotateGestureDetector.isEnabled.also {
          gesturesManager.rotateGestureDetector.isEnabled = false
        }
      }
```

Or if we want to define the specific logic when certain type of gestures are handled, I guess we can rename to `handleGestureStart(gesture: Type)` and `handleGestureEnd(gesture: Type)`

Reply:
isn't `Shove`gesture is different than `ScaleQuickZoom`/ `DoubleTapToZoom`?
 iirc shove is used for pitching of the map while DoubleTap or ScaleQuick is for zoom.

Reply:
Another idea I had for this was to have a stack of remembered settings, and pop them when the operation ends.
Could there be a case where multiple operations could be happening simultaneously @pengdev ?  In this case, we might have a race condition for restoring settings in the correct order :/

Reply:
> In this case, we might have a race condition for restoring settings in the correct order

Should be a problem, since all gestures happen on main thread, thus if some gesture is disabled it means it wont send updates anymore.